packages/keys/src/base58.ts

```diff
@@ -1,10 +1,12 @@
 import bs58 from 'bs58';
 
-export type Base58EncodedAddress = string & { readonly __base58EncodedAddress: unique symbol };
+export type Base58EncodedAddress<TAddress extends string = string> = TAddress & {
+    readonly __base58EncodedAddress: unique symbol;
+};
 
 export function assertIsBase58EncodedAddress(
     putativeBase58EncodedAddress: string
```

Comment:
Any reason to not also make this generic over `<TAddress extends string = string>`? Or does that encompass potentially too many types?

Reply:
It actually already is; it's sort of magic.

<img width="673" alt="image" src="https://github.com/solana-labs/solana-web3.js/assets/13243/eb188229-787a-4054-9b59-d14f1fec05dd">

![anna-kendrick-mind-blown](https://github.com/solana-labs/solana-web3.js/assets/13243/c9ddb3ab-140f-4254-8080-2298ff576a02)

Reply:
dang, that's really cool